cmake/external/onnxruntime_external_deps.cmake

```diff
@@ -335,6 +335,7 @@ if(onnxruntime_USE_CUDA)
     URL ${DEP_URL_microsoft_gsl}
     URL_HASH SHA1=${DEP_SHA1_microsoft_gsl}
     PATCH_COMMAND ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/gsl/1064.patch
+    FIND_PACKAGE_ARGS 4.0 NAMES Microsoft.GSL
```

Comment:
> If you got GSL from the system, most likely it didn't apply the patch that is ONNX Runtime specific.

I would actually love to have `FIND_PACKAGE_ARGS` in both if-else branches, so that one could prevent CMake from attempting to access the network if they wanted to. E.g. in nixpkgs we're happy to apply the patch ourselves. Also CC https://github.com/microsoft/onnxruntime/pull/13573/files#r1371748420 on whether the branching is necessary?

What do you think? @bverhagen @snnn

Reply:
So you mean this change is good and we should just take it?  I agree on that.